### PR TITLE
change Kafka Topic resource behaviour 

### PR DESCRIPTION
**Pull request by ivan-savciuc**
change Kafka Topic resource `retention_hours` behaviour according to API specification

### CODE REVIEW

**Comment by ivan-savciuc** [comment]
Ok I see, then I'll change that as well

**Comment by ivan-savciuc** [comment]
Updated PR using latest go-client with `retention_hours` as an optional field.

File: aiven/resource_kafka_topic.go [hunk 1]
```diff
@@ -46,10 +47,25 @@ var aivenKafkaTopicSchema = map[string]*schema.Schema{
...
 	},
 	"retention_hours": {
-		Type:        schema.TypeInt,
-		Optional:    true,
-		Default:     72,
-		Description: "Retention period (hours)",
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Description:  "Retention period (hours)",
+		ValidateFunc: validation.IntBetween(0, 2562047788015),
```
Comment (rikonen) [inline aiven/resource_kafka_topic.go:53]: -1 is valid value, means infinite retention

File: aiven/resource_kafka_topic.go [hunk 2]
```diff
@@ -46,10 +47,25 @@ var aivenKafkaTopicSchema = map[string]*schema.Schema{
...
+		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+			// When a retention hours field is not set to any value and consequently is null (empty string).
+			// Allow ignoring those.
+			if new == "" {
+				return true
+			}
+
+			// When a retention hours field is set to -1 it means that it not defined.
+			// Allow ignoring those for gracefully transition to the new logic.
+			if old == "-1" {
```
Comment (rikonen) [inline aiven/resource_kafka_topic.go:63]: This doesn't look correct as -1 doesn't mean not defined.

File: aiven/resource_kafka_topic.go [hunk 3]
```diff
@@ -159,9 +175,16 @@ func resourceKafkaTopicRead(d *schema.ResourceData, m interface{}) error {
 	if err := d.Set("retention_bytes", topic.RetentionBytes); err != nil {
 		return err
 	}
-	if err := d.Set("retention_hours", topic.RetentionHours); err != nil {
-		return err
+
+	if topic.RetentionHours != nil {
+		// it could be -1, which means that this field is not defined.
```
Comment (rikonen) [inline aiven/resource_kafka_topic.go:180]: Means retention time is infinite

**Review by rikonen — CHANGES_REQUESTED** [review]
-1 handling doesn't look correct

**Review by rikonen — APPROVED** [review]

**Pull request merged**